### PR TITLE
Allow downloading of all columns in the NASA Exoplanet Archive

### LINKED ISSUE #1027: Additional columns in NASA Exoplanet Archive queries
When querying the NASA Exoplanet Archive, the standard columns returned from the database table do not include some important exoplanet parameters, such as transit midpoint (the default columns are listed [here](https://exoplanetarchive.ipac.caltech.edu/docs/API_exoplanet_columns.html#defaultcol). **It would be useful if we could include additional columns to the query**. However, the Exoplanet Archive's API seems to allow only one additional column (with the URL string '&addcol'), which is not very useful if more than one additional column is required.

I think the best solution would be to have a standard list of columns that should be returned from the query, and then the user should be able to add any column they want. Then, it should be as simple as using the URL string '&select=column0,column1,column3...'.

### PR DESCRIPTION
**Pull request by mwcraig**
This PR fixes #1027 by adding an option to get all columns (instead of just the default set) from the NASA Exoplanet Archive.

One test will fail; if I download with `all_columns=True` first, then `False` fails. I assume there is some cacheing going on that I haven't been able to stop.

Arguably it would make more sense just to make it *always* return all columns (though @bmorris3 may know of a reason not too); then we would only need one test :).

This catalog is more complete than than the Exoplanet Orbit Database, so it would be nice to have it return more transit-related parameters.

### CODE REVIEW

File: astroquery/nasa_exoplanet_archive/nasa_exoplanet_archive.py [hunk 1]
```diff
@@ -104,6 +112,9 @@ def query_planet(self, planet_name, table_path=None):
         table_path : str (optional)
             Path to a local table file. Default `None` will trigger a
             download of the table from the internet.
+        all_columns : bool (optional)
+            Return all available columns. The default returns only the
+            columns in the default category at the link above.
```
Comment (bsipocz) [inline astroquery/nasa_exoplanet_archive/nasa_exoplanet_archive.py]: could you add an extra empty line here, the docstring doesn't render correctly atm.

File: astroquery/nasa_exoplanet_archive/nasa_exoplanet_archive.py [hunk 2]
```diff
@@ -58,14 +58,21 @@ def get_confirmed_planets_table(self, cache=True, show_progress=True,
...
         Returns
```
Comment (bsipocz) [inline astroquery/nasa_exoplanet_archive/nasa_exoplanet_archive.py]: could you add an extra empty line here, the docstring doesn't render correctly atm.

**Comment by bsipocz** [comment]
@mwcraig - Could you also add a changelog entry?

**Comment by mwcraig** [comment]
@bsipocz -- thanks, have addressed the comments. How do recommend proceeding regarding the failing test? Looks like I either need to clear the cache between tests or switch to always returning all columns....

**Comment by bsipocz** [comment]
@mwcraig - There is the `cache` kwarg is ignored then it's a bug on its own right. Could you open an issue for it please?. But it's beyond this PR, so feel free to move your new test to the bottom of the tests, then arguably everything should still pass.

**Comment by mwcraig** [comment]
Found and fixed the cacheing error. The class was doing its own cacheing by checking whether `self._table` existed; since `NasaExoplanetArchive` is an instance of the class, `self._table` is set the first time `get_confirmed_planets_table` is called and stays the same for all tests, so the first test was setting the table contents and ignoring the cache setting.

**Comment by bsipocz** [comment]
Thanks @mwcraig!

**Pull request merged**